core/arch/arm/mm/core_mmu_v7.c

```diff
@@ -502,7 +503,10 @@ bool core_mmu_divide_block(struct core_mmu_table_info *tbl_info,
 		return false;
 
 	entry = (uint32_t *)tbl_info->table + idx;
-	assert(get_desc_type(1, *entry) == DESC_TYPE_SECTION);
+	assert(!*entry || get_desc_type(1, *entry) == DESC_TYPE_SECTION);
+
+	/* We need to flush TLBs only if there already was some mapping */
+	flush_tlb = *entry;
```

Comment:
If `*entry` contains some value, it can overflow the `bool`. Better normalize with `!!` before assingment.